### PR TITLE
feat(notifications): Confirmation Modal

### PR DESCRIPTION
**Pull request by mgaeta**
The new Notification Settings designs treat the default settings as an override when the value is "off". This override erases all of the user's fine-tuned settings so this PR adds a confirmation modal.
Screen Shot 2021-09-02 at 12 19 44 PM

### CODE REVIEW

File: static/app/views/settings/components/forms/selectField.tsx [hunk 1]
```diff
@@ -79,7 +84,43 @@ export default class SelectField<
...
+    const {allowClear, confirm, multiple, small, ...otherProps} = this.props;
+
+    if (confirm) {
+      return (
+        <InputField
+          {...otherProps}
+          alignRight={small}
+          field={({onChange, onBlur, required: _required, ...props}) => (
+            <Confirm
+              message={confirm.message}
```
Comment (mgaeta) [inline static/app/views/settings/components/forms/selectField.tsx:96]: I tried to copy `confirm` from `BooleanField`, but I can't figure out how to modulate the `message` with the new value.

**Comment by ceorourke** [comment]
I tried to find another instance of a confirmation modal in the app:

Screen Shot 2021-07-26 at 4 17 35 PM

What do you think about making it a little more like this (I'm trying to come up with text that makes sense on both the main page where you don't see that it's a "default" as well as on the fine tuning page) where you have a heading text that's bold and more information text that's not bold:

**Are you sure you want to disable these notifications?**
Turning this off will irreversibly overwrite all of your fine-tuning settings to "off".

**Comment by ceorourke** [comment]
When I turn my settings off and then try to turn them back on, it fails with "Cannot read property 'toString' of undefined"

**Comment by mgaeta** [comment]
@NisanthanNanthakumar can you rebase this? #28144 did something

**Comment by mgaeta** [comment]
Took a look but this is a frontend issue. The value being passed to the backend is 
```json
{
  "alerts": {
	"me": {
		"email": undefined
	}
}
```

**Comment by mgaeta** [comment]
Updated the screenshot.

**Review by ceorourke — APPROVED** [review]

**Pull request merged**